examples/bsp_ext.py

```diff
@@ -0,0 +1,67 @@
+import os
+from pathlib import Path
+try:
+    import ruamel.yaml
+except ImportError:
+    print("!! ruamel.yaml package is not installed. No BSP extension is added !!")
+    quit()
+
+
+def action_extensions(base_actions, project_path=os.getcwd()):
+    """ Describes extension for Board Support Packages. """
+
+    def bsp_short_name(bsp):
+        return bsp.split('/')[-1]
+
+    def set_bsp_callback(action: str, ctx, args, **kwargs: str) -> None:
+        yaml = ruamel.yaml.YAML()
+        # List of supported BSPs and their targets
+        # Don't forget to update this dictionary and build_example_for_all_bsps.sh when adding new BSP
+        bsps = {
+            'esp_wrover_kit'       : 'esp32',
+            'esp32_azure_iot_kit'  : 'esp32',
+            'esp32_s2_kaluga_kit'  : 'esp32s2',
+            'esp32_s3_eye'         : 'esp32s3',
+            'esp32_s3_lcd_ev_board': 'esp32s3',
+            'esp32_s3_usb_otg'     : 'esp32s3',
+            'esp-box'              : 'esp32s3',
+            'esp32_s3_korvo_2'     : 'esp32s3',
+            }
+        manifest_path  = Path(project_path) / 'main' / 'idf_component.yml'
+        with open(manifest_path, 'r+') as f:
+            manifest = yaml.load(f)
+
+            # Remove all BSPs
+            for dep in list(manifest['dependencies']):
+                if bsp_short_name(dep) in bsps:
+                    del manifest['dependencies'][dep]
+
+            # Add one we need
+            manifest['dependencies'].insert(0, kwargs['bsp'], {'version': '*', 'override_path': ('../../../' + bsp_short_name(kwargs['bsp']))})
+
+            # Save file
+            f.truncate(0)
+            f.seek(0)
+            yaml.dump(manifest, f)
```

Comment:
`ruamel` supports `Path` objects directly, so it can be as simple as:
```suggestion
        manifest = yaml.load(manifest_path)

        # Remove all BSPs
        for dep in list(manifest['dependencies']):
            if bsp_short_name(dep) in bsps:
                del manifest['dependencies'][dep]

        # Add one we need
        manifest['dependencies'].insert(0, kwargs['bsp'], {'version': '*', 'override_path': ('../../../' + bsp_short_name(kwargs['bsp']))})

        yaml.dump(manifest, manifest_path)
```